src/OpenTok/OpenTok.php

```diff
@@ -35,26 +35,47 @@ class OpenTok
     private $apiSecret;
     /** @internal */
     private $client;
-    /** @internal */
+
+    /**
+     * @var array
+     * @internal
+     * Options that can override the defaults. Additionally, you can set the keys
+     * application_id & private_key_path to strings that will then override the default
+     * OpenTok Client behaviour when making requests to the Vonage Video API.
+     */
     public $options;
 
     /** @internal */
     public function __construct($apiKey, $apiSecret, $options = array())
     {
+        $validateKeys = true;
+        Validators::validateVonageJwtArguments($options);
```

Comment:
We should do this on `$apiKey` and `$apiSecret` and not the additional options. Then if the validation fails drop it back to Opentok authentication

Reply:
Does this mean the shim is actually going to automatically push people to Vonage Video as the default? I thought it was "here is an option to migrate if you want"

Reply:
The shim will use whatever credentials are passed in. This means that an OT user will pass in their Key/Secret, and a Vonage user will pass in an AppID/Private Key. The SDK will then switch the base URL, auth, and client token generation to Vonage if it's passed Vonage credentials.

There is no "default" like with the JWT generation (Defaulting to JWT but allowing a T1 fallback), it's all based on how you pass in your credentials.

This is also meant to be a drop-in solution, so all a user has to do is update and then change their creds. Passing additional flags as an option breaks that commitment.